package/kedro_viz/data_access/repositories/catalog.py

```diff
@@ -55,8 +56,9 @@ def get_dataset(self, dataset_name: str) -> Optional[AbstractDataSet]:
                 dataset_obj = self._catalog._get_dataset(dataset_name, suggest=False)
             else:  # pragma: no cover
                 dataset_obj = self._catalog._get_dataset(dataset_name)
+        # if dataset has no catalog entry, it is a MemoryDataSet
         except DataSetNotFoundError:  # pragma: no cover
```

Comment:
Wonder if this should be covered in tests, given that it does make a functional difference.

Reply:
yes, i will write the tests now !